pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -242,6 +248,28 @@ func (c *CatalogSource) Poll() bool {
 	return true
 }
 
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) (err error) {
+	var updateStrategyMap map[string]map[string]string
+	if err = json.Unmarshal(data, &updateStrategyMap); err != nil {
+		return err
+	}
+	registryPoll := &RegistryPoll{}
+	duration, err := time.ParseDuration(updateStrategyMap["registryPoll"]["interval"])
+	if err != nil {
+		registryPoll.ParsingError = fmt.Sprintf("error parsing spec.updateStrategy.registryPoll.interval. Setting the default value of %v. Error:%v", DefaultRegistryPollDuration, err)
+		defaultTime, parseErr := time.ParseDuration(DefaultRegistryPollDuration)
```

Comment:
Should we be setting a default time? If this is not set correctly, should we just default to stripping instead?

Reply:
I don't have a strong preference either way. To me either approach let's users know there's been some issue. Setting a default time does take care of the scenario where the CatalogSource was added by a ci and the admin was expecting to get updates anyway by default (instead of going days/months without looking at the error and expecting updates to come in, and then the issue getting escalated when realization sets in that updates were not coming in).

Reply:
I'm inclined with treating it as a terminal error when reconciling the catalog source and clearly stating on the status the error in parsing the interval. Other behaviors like setting a default, or removing the interval, could be surprising for users, versus the immediate feedback when the catalog spec is invalid.